backend/src/providers/LocalVMProvider.ts

```diff
@@ -41,9 +41,40 @@ export default class LocalVMProvider implements InstanceProvider {
     return this.availableInstances.get(identifier);
   }
 
-  async createServer(): Promise<VMEndpoint> {
+  async createServer(identifier: string): Promise<VMEndpoint> {
     if (this.availableInstancesList.length > 0) {
-      return this.availableInstances.get(this.availableInstancesList.pop());
```

Comment:
I suggest we extract the new capability into a new Provider. This preserves the simple logic in this one and allows injecting the more complex one on demand as a dependency.

Reply:
good point! I pushed a draft for that...